### PR TITLE
Add Support for declaring subviews

### PR DESCRIPTION
**Pull request by nicknisi**

### CODE REVIEW

File: thumbs.js [hunk 1]
```diff
@@ -98,6 +98,57 @@
...
+                if (a) {
+                    _.each(a.split(","), function (arg) {
+                        arg = arg.split(":");
+                        args[arg[0]] = arg[1];
+                    }, this);
+                }
+
+                _.extend(args, { el: this});
+
+                if (v.indexOf("/") >= 0 && typeof require === "function") {
```
Comment (doug-martin) [inline thumbs.js:142]: We have the `node.js` case to account for also

File: thumbs.js [hunk 2]
```diff
@@ -98,6 +98,57 @@
...
+                    require([v], function (View) {
+                        new View(args).render();
+                    });
+                } else if (v.indexOf(".") >= 0) {
+                    // assume that this is a global path
+                    eval("new "+v+"(args).render();");
```
Comment (doug-martin) [inline thumbs.js]: I think if we get here we might want to not use eval but instead look up the object in window or global depending on server or DOM

So a simple look up could be

```
var getObject = _.memoize(function getObject(name){
   var o = window || global;
   _.each(name.split("."), function(part){
      o = o[part];
   });
  return o;
});
```

and to use

```
var Const = getObject(v);
if("function" === typeof Const){
   new Const(args).render();
}
```

File: thumbs.js [hunk 3]
```diff
@@ -98,6 +98,57 @@
     Collection = thumbs.Collection = Collection.extend(_super);
     Router = thumbs.Router = Router.extend(_super);
 
+    var Subview = {
+        _subviews: null,
+        initialize: function () {
+            this._super('initialize', arguments);
+            this._subviews = [];
```
Comment (doug-martin) [inline thumbs.js:106]: You started to keep track but didnt I think we need to keep a reference to the sub view and call remove when the parent is removed....do you agree?

File: thumbs.js (hunk 1 above)
Comment (nicknisi) [inline thumbs.js:142]: In what case would I be trying to run Backbone on the server?

Comment (doug-martin) [inline thumbs.js:142]: Well Backbone supports it I think people use it in combination with `jsdom`, however its just determining whether or not require is AMD or commonjs.

**Pull request closed without merging**